internal/ethapi/api.go

```diff
@@ -1170,13 +1154,15 @@ func DoEstimateGas(ctx context.Context, b Backend, args TransactionArgs, blockNr
 	} else {
 		feeCap = common.Big0
 	}
+
+	st, header, err := b.StateAndHeaderByNumberOrHash(ctx, blockNrOrHash)
```

Comment:
had to use "st" instead of "state" for this var to avoid name conflict with type state.StateDB referenced below

Reply:
I'd probably just pull the `execute` function out into something like `executeEstimate` since you're giving up on expressiveness of variable name to appease the compiler.

Reply:
done